Improve db profiling facilities

ttung opened the pull request:
### Description
1. Add the ability to disable profiling.
2. Add a contextmanager that turns on and off profiling within a region of code.

### Test plan
Used it to profile /api/phylo_trees, seems to be n_queries=2

ttung commented:
Manually enabled it #327, and got this output:

```
DEBUG    sqltime:connection.py:78 Start Query: SELECT aspen.users.id AS aspen_users_id, aspen.users.name AS aspen_users_name, aspen.users.email AS aspen_users_email, aspen.users.auth0_user_id AS aspen_users_auth0_user_id, aspen.users.group_admin AS aspen_users_group_admin, aspen.users.system_admin AS aspen_users_system_admin, aspen.users.agreed_to_tos AS aspen_users_agreed_to_tos, aspen.users.group_id AS aspen_users_group_id, groups_1.id AS groups_1_id, groups_1.name AS groups_1_name, groups_1.address AS groups_1_address, can_see_1.id AS can_see_1_id, can_see_1.viewer_group_id AS can_see_1_viewer_group_id, can_see_1.owner_group_id AS can_see_1_owner_group_id, can_see_1.data_type AS can_see_1_data_type 
FROM aspen.users LEFT OUTER JOIN aspen.groups AS groups_1 ON groups_1.id = aspen.users.group_id LEFT OUTER JOIN aspen.can_see AS can_see_1 ON groups_1.id = can_see_1.viewer_group_id 
WHERE aspen.users.auth0_user_id = %(auth0_user_id_1)s
DEBUG    sqltime:connection.py:83 Query Complete!
DEBUG    sqltime:connection.py:84 Total Time: 0.005229
DEBUG    sqltime:connection.py:78 Start Query: SELECT anon_1.aspen_workflows_id AS anon_1_aspen_workflows_id, anon_1.aspen_workflows_workflow_type AS anon_1_aspen_workflows_workflow_type, anon_1.aspen_workflows_start_datetime AS anon_1_aspen_workflows_start_datetime, anon_1.aspen_workflows_end_datetime AS anon_1_aspen_workflows_end_datetime, anon_1.aspen_workflows_workflow_status AS anon_1_aspen_workflows_workflow_status, anon_1.aspen_workflows_software_versions AS anon_1_aspen_workflows_software_versions, anon_1.aspen_phylo_runs_workflow_id AS anon_1_aspen_phylo_runs_workflow_id, anon_1.aspen_phylo_runs_group_id AS anon_1_aspen_phylo_runs_group_id, anon_1.aspen_phylo_runs_template_file_path AS anon_1_aspen_phylo_runs_template_file_path, anon_1.aspen_phylo_runs_template_args AS anon_1_aspen_phylo_runs_template_args, (SELECT count(%(count_2)s) AS count_1 
FROM aspen.samples JOIN aspen.phylo_tree_samples ON aspen.samples.id = aspen.phylo_tree_samples.sample_id JOIN (aspen.entities JOIN aspen.phylo_trees ON aspen.entities.id = aspen.phylo_trees.entity_id) ON aspen.phylo_tree_samples.phylo_tree_id = aspen.phylo_trees.entity_id 
WHERE aspen.entities.producing_workflow_id = anon_1.aspen_phylo_runs_workflow_id) AS phylo_run_genome_count, entities_1.id AS entities_1_id, entities_1.entity_type AS entities_1_entity_type, entities_1.producing_workflow_id AS entities_1_producing_workflow_id 
FROM (SELECT aspen.workflows.id AS aspen_workflows_id, aspen.workflows.workflow_type AS aspen_workflows_workflow_type, aspen.workflows.start_datetime AS aspen_workflows_start_datetime, aspen.workflows.end_datetime AS aspen_workflows_end_datetime, aspen.workflows.workflow_status AS aspen_workflows_workflow_status, aspen.workflows.software_versions AS aspen_workflows_software_versions, aspen.phylo_runs.workflow_id AS aspen_phylo_runs_workflow_id, aspen.phylo_runs.group_id AS aspen_phylo_runs_group_id, aspen.phylo_runs.template_file_path AS aspen_phylo_runs_template_file_path, aspen.phylo_runs.template_args AS aspen_phylo_runs_template_args 
FROM aspen.workflows JOIN aspen.phylo_runs ON aspen.workflows.id = aspen.phylo_runs.workflow_id) AS anon_1 LEFT OUTER JOIN aspen.entities AS entities_1 ON anon_1.aspen_workflows_id = entities_1.producing_workflow_id 
WHERE anon_1.aspen_phylo_runs_group_id IN (SELECT anon_2.id 
FROM (SELECT aspen.groups.id AS id 
FROM aspen.groups JOIN aspen.users ON aspen.groups.id = aspen.users.group_id 
WHERE aspen.users.auth0_user_id = %(auth0_user_id_1)s) AS anon_2) AND anon_1.aspen_workflows_workflow_status = %(aspen_workflows_workflow_status_1)s
DEBUG    sqltime:connection.py:83 Query Complete!
DEBUG    sqltime:connection.py:84 Total Time: 0.005616
```

phoenixAja reviewed (approved):
looks good, can you add in the docs how to toggle this?

Commit:
Improve db profiling facilities

### Description
1. Add the ability to disable profiling.
2. Add a contextmanager that turns on and off profiling within a region of code.

### Test plan
Used it to profile /api/phylo_trees, seems to be n_queries=2

ttung commented:
updated docs.
